packages/@lwc/engine-core/src/framework/base-bridge-element.ts

```diff
@@ -121,9 +142,25 @@ export function HTMLBridgeElementFactory(
     const { attributeChangedCallback: superAttributeChangedCallback } = SuperClass.prototype as any;
     const { observedAttributes: superObservedAttributes = [] } = SuperClass as any;
     const descriptors: PropertyDescriptorMap = create(null);
+
+    // present a hint message so that developers are aware that they have not decorated property with @api
+    if (process.env.NODE_ENV !== 'production') {
+        for (const propName in protoDescriptors) {
+            const propDescriptor = protoDescriptors[propName];
+            if (
+                (propDescriptor.get || propDescriptor.set) &&
```

Comment:
Why is this condition needed? I removed it, and the tests still passed.

Reply:
I remember seeing some props like "constructor"

Reply:
Ah yeah, I see that this often not only contains `constructor`, but also `render`, `connectedCallback`, `errorCallback`, `disconnectedCallback`, `renderedCallback`, etc.

I can see two options here:

1. Have a hard-coded blocklist of methods we don't want to warn on (e.g. `constructor`)
2. Don't do anything special

#<span></span>2 might be best, because it seems pretty unlikely that someone will accidentally call `elm.renderedCallback()` (or whatever) from the outside. And even if they do, apparently [you can add `@api` to it and it actually works](https://stackblitz.com/edit/salesforce-lwc-nfxbjb?file=src%2Fmain.js). 😆

Interestingly though, you can't add `@api` to `constructor` because [Babel blocks it](https://github.com/babel/babel/blob/257c06457f3e219d8fe90586937354b566a99476/packages/babel-parser/src/parse-error/standard-errors.ts#L64).

So maybe just adding special logic to exclude `constructor` is sufficient here.

Reply:
I decided to go with 2.